lib/maths/CSeasonalTime.cc

```diff
@@ -27,6 +27,7 @@ namespace {
 // DO NOT change the existing tags if new sub-classes are added.
 const core::TPersistenceTag DIURNAL_TIME_TAG{"a", "diurnal_time"};
 const core::TPersistenceTag ARBITRARY_PERIOD_TIME_TAG{"b", "arbitrary_period_time"};
+const std::string DELIMITER(",");
```

Comment:
There's a subtle dependency between this and the `DELIMITER` and `PAIR_DELIMITER` in `core::CPersistUtils`.  This one has to be different to those two, since restoration is searching for it starting at the beginning of a string whose beginning portion contains `core::CPersistUtils::DELIMITER` and `core::CPersistUtils::PAIR_DELIMITER`.  So this could be commented.  Alternatively `rfind` could be used instead of `find` when deconstructing the persisted string.

Reply:
I like the rfind idea. I thought about defining both here to make this explicit, but decided this unit test would fail if anyone changed the delimiter used by CPersistUtils. (On that front, this would also invalidate all our saved state so I don't see it ever happening, but better to be safe.)